backend/actions/transaction_list.js

```diff
@@ -11,28 +11,138 @@ app.use((req, res, next) => {
   next();
 });
 
-app.get('/transactions', (req, res) => {
-  // List transactions 100 to 120
-  // bitcoin-cli listtransactions "*" 20 100
-  // bitcoin-cli getchaintxstats
+async function getBlockchainInfo() {
+  const result = await cl.getBlockchainInfo();
+  return result.headers;
+}
+
+async function getTxCount() {
+  const result = await cl.getChainTxStats();
+  return result.txcount;
+}
+
+async function getBlockWithTx(blockNum) {
+  const blockHash = await cl.getBlockHash(blockNum);
+  const result = await cl.getBlock(blockHash);
+  return result;
+}
+
+async function getMemTx() {
+  const result = await cl.getRawMempool();
+  let list = [];
+  list  = result.map( (tx) => {
+    const response = cl.command([
+      {
+        method: 'getrawtransaction', 
+        parameters: {
+          txid: tx,
+          verbose : true
+        }
+      }
+    ])
+    return response;
+  }) 
+
+  const promiseArray = await Promise.all(list);
+  const memTxArray = promiseArray.map((list) => list[0]);
+
+  const memEntryArray = memTxArray.map( (trans) => {
+    const response = cl.command([
+      {
+        method: 'getmempoolentry', 
+        parameters: {
+          txid: trans.txid,
+        }
+      }
+    ])
+    return response;
+  })
+
+  const entryPromiseArray = await Promise.all(memEntryArray)
+  const finalArray = memTxArray.map( (trans, idx) => {
+    trans.time = entryPromiseArray[idx][0].time
+    return trans;
+  })
+
+  return finalArray.sort((a,b) => b.time - a.time);
+}
+
+app.get('/transactions', async (req, res) => {
+  //Return a List of transactions
+
   var perPage = Number(req.query.perPage);
   var page = Number(req.query.page);
 
-  cl.command([
-    { 
-      method: 'listtransactions', 
-      parameters: {
-        dummy: "*",
-        count: perPage,
-        skip: page*perPage
+  getTxCount().then( (txCount) => {
+    
+    getBlockchainInfo().then( async (bestBlockHeight) => {
+
+      let count = 0, transList= [], overheadTxCount = 0;
+      const memTxList = await getMemTx();
+      if(( memTxList.length) > (perPage*(page-1))){
+
+        let j = (perPage*(page-1));
+        while(j < memTxList.length){ 
+          let amount = 0;
+          memTxList[j].vout.forEach( (vout) => {amount += vout.value})
+          memTxList[j].amount = amount;
+          memTxList[j].confirmations = 0;
+          transList.push(memTxList[j]);
+          j++;
+          count++;
+          if(count == perPage){
+            break;
+          }
+        }
+      }
+
```

Comment:
Thank you for the review. We can use slice on `memTxList` if `( memTxList.length) > (perPage*(page))` but we'd still have to iterate through memTxList for [amount](https://github.com/chaintope/tapyrus-explorer/pull/18/files/16c778eb8e316f303b2904a4f239617ff4bfa317#diff-080aeb14685199a336253e972c227a28R88) and confirmations. 

`getBlockWithTx` is needed to get the blocks, I don't know how it could be removed depending on `perPage*(page)` or `count`.

Reply:
> Can we skip this while block (and calling getBlockWithTx) if count == perPage ?

In other words, 
Is `getBlockWithTx` needed even if memTxList has perPage records?

Reply:
I checked this part is updated at new commit d8db8ed. look good to me.